enterprise/server/remote_execution/snaploader/snaploader.go

```diff
@@ -583,14 +584,23 @@ func (l *FileCacheLoader) cacheCOW(ctx context.Context, name string, remoteInsta
 				}
 			}
 
+			// If the chunk was mapped but the lazy digest is not set, it could not
+			// have been fetched remotely, because we use the lazy digest to do so.
+			// This means we must have generated the chunk from a local file
+			// Note: This has to be called before we call c.Digest(), which sets
+			// the lazy digest
+			chunkFetchedRemotely := c.Mapped() && c.LazyDigest() != nil
```

Comment:
Shouldn't the lazyDigest be non-nil if we fetched the chunk from local filecache too?

also nit: maybe clearer/more explicit to add a function to the chunk like `WasFetchedRemotely()`

Reply:
Hmmm yeah. I guess we could hit a situation where the chunk is in the local cache but gets evicted remotely and we'd still want it to be uploaded the remote cache. 

I'll switch to your suggestion of marking where the chunk is from. We could use this data in a metric, to validate our hypothesis that most chunks should be cached locally

Reply:
If we fetched the chunk from local filecache, we still shouldn't need to re-cache it right? (The only thing we might want to do is update the atime)

Either way, I like your suggestion of explicitly marking where the chunk is from. We could use this data in a metric to validate our hypothesis that most chunks should be cached locally

Reply:
I was mostly confused by the `chunkFetchedRemotely` naming here, but yeah re-caching is also somewhat of a concern. I guess we don't really need to update the atime since GetTree should handle it, and we call that unconditionally?

Reply:
Actually I think we should be okay. If we read it from the local cache, that will update its atime. If we read it from the remote cache, we write it to the local cache